src/NLog/Internal/ReflectionHelpers.cs

```diff
@@ -92,6 +92,72 @@ public static bool IsStaticClass(this Type type)
         {
             return type.IsClass && type.IsAbstract && type.IsSealed;
         }
+
+        /// <summary>
+        /// Optimized delegate for calling MethodInfo
+        /// </summary>
+        /// <param name="target">Object instance, use null for static methods.</param>
+        /// <param name="arguments">Complete list of parameters that matches the method, including optional/default parameters.</param>
+        /// <returns></returns>
+        public delegate object LateBoundMethod(object target, object[] arguments);
+
+        /// <summary>
+        /// Creates an optimized delegate for calling the MethodInfo using Expression-Trees
+        /// </summary>
+        /// <param name="methodInfo">Method to optimize</param>
+        /// <returns>Optimized delegate for invoking the MethodInfo</returns>
+        public static LateBoundMethod CreateLateBoundMethod(MethodInfo methodInfo)
+        {
+            // parameters to execute
+            var instanceParameter = Expression.Parameter(typeof(object), "instance");
+            var parametersParameter = Expression.Parameter(typeof(object[]), "parameters");
+
+            // build parameter list
+            var parameterExpressions = new List<Expression>();
+            var paramInfos = methodInfo.GetParameters();
+            for (int i = 0; i < paramInfos.Length; i++)
+            {
+                // (Ti)parameters[i]
+                var valueObj = Expression.ArrayIndex(parametersParameter, Expression.Constant(i));
+
+                Type parameterType = paramInfos[i].ParameterType;
+                if (parameterType.IsByRef)
+                    parameterType = parameterType.GetElementType();
+
+                var valueCast = Expression.Convert(valueObj, parameterType);
+
+                parameterExpressions.Add(valueCast);
+            }
+
+            // non-instance for static method, or ((TInstance)instance)
+            var instanceCast = methodInfo.IsStatic ? null :
+                Expression.Convert(instanceParameter, methodInfo.ReflectedType);
```

Comment:
@snakefoot 

another netstandard challenge, `methodInfo.ReflectedType` isn't there. Any idea how to fix it? I dunno what the `ReflectedType` is in this case.